api/user/schema.py

```diff
@@ -112,11 +113,46 @@ def mutate(self, info, email, **kwargs):
         exact_user = query_user.filter(UserModel.email == email).first()
         if not exact_user:
             raise GraphQLError("User not found")
-        user_role = UsersRole.query.filter_by(user_id=exact_user.id).first()
-        new_role = kwargs.pop('role_id')
-        user_role.role_id = new_role
-        with SaveContextManager(user_role, 'Role id', new_role):
-            return ChangeUserRole(user=exact_user)
+
+        if not exact_user.roles:
+            raise GraphQLError('user has no role')
+
+        new_role_obj = RoleModel.query.filter_by(id=kwargs['role_id']).first()
```

Comment:
What does 'new_role_obj' is supposed to represent here? Avoid abbreviations in naming your variables.

Reply:
noted. I have changed the variable to `new_role` which is meant to hold the new role queried from the `roles` table